### PR TITLE
Implement censoring words on server

### PR DESCRIPTION
**Pull request by def-**

### CODE REVIEW

**Comment by edg-l** [comment]
Why on the server? Make it on the client and optional please.

File: src/game/server/gamecontext.h [hunk 1]
```diff
@@ -73,6 +76,7 @@ class CGameContext : public IGameServer
 	CNetObjHandler m_NetObjHandler;
 	CTuningParams m_Tuning;
 	CTuningParams m_aTuningList[NUM_TUNEZONES];
+	array<string> m_aCensorlist;
```
Comment (Learath2) [inline src/game/server/gamecontext.h:79]: Might aswell make this a sorted_array and binary search, should be no penalty for a short list but can support a massive list much easily

Comment (def-) [reply src/game/server/gamecontext.h:79]: Why? How can I binary search for the censored string?

**Comment by heinrich5991** [comment]
bors r+

Thanks.

**Pull request merged**